js/main.js

```diff
@@ -7,45 +9,82 @@ const $anchorEntries = document.querySelector('.entriesBtn');
 const $entryForm = document.querySelector('.query_one');
 const $entriesBottom = document.querySelector('.query_two');
 const $newBtn = document.querySelector('.new_js');
+const $nullMsg = document.querySelector('.nullMsg');
+const $findingTheDomCreatedLis = document.querySelector('li.row');
+const $eHeader = document.querySelector('.entryHeader');
+
+// event on the photoURL input which listen's for when the user type's
 
 $photoId.addEventListener('input', (e) => {
   $imgSrcChange.setAttribute('src', e.target.value);
 });
+
 // submit event
 
+// Listner method for the actual form submition
+
 $form.addEventListener('submit', (e) => {
   e.preventDefault();
+
+  // This conditional checks if the data.editing pencil is a value of null or not null
+
   const objStorageFormInputs = {
     entryId: data.nextEntryId,
     title: e.target.elements.title.value,
     photoID: e.target.elements.photoID.value,
     textarea: e.target.elements.textarea.value,
   };
+  if (data.editing === null) {
+    data.nextEntryId++;
+
+    data.entries.unshift(objStorageFormInputs);
+
+    $imgSrcChange.setAttribute('src', './images/placeholder-image-square.jpg');
 
-  data.nextEntryId++;
+    $form.reset();
 
-  data.entries.unshift(objStorageFormInputs);
+    // render the renderEntry callback on the form if the form's submitted
 
-  $imgSrcChange.setAttribute('src', './images/placeholder-image-square.jpg');
+    const $callRender = renderEntry(objStorageFormInputs);
+    // Update for the editing view using a condition
+    // Current step in code-journal
 
-  $form.reset();
+    $entriesList.prepend($callRender);
+  } else if (data.editing !== null) {
+    // Looping over the Data.entrie's array and finding which element's matche's the data.editing
+    data.entries.forEach((dataEl) => {
+      if (dataEl === data.editing) {
+        // Replacing the object in the data.entries object/array with the data.editing object
+        // data.entries = data.editing
+        // Updating the objStorageFoemInputs.entryId with editing value inside the object's array in LS
+        objStorageFormInputs.entryId = data.editing;
+        // DOM tree callback call with the object arg.
+        data.entries = data.editing;
+        // })
 
-  // render the renderEntry callback on the form if the form's submitted
+        // Finding the existing LI element and changing/replacing it's value to
+        // to match the editing array
 
-  const $callRender = renderEntry(objStorageFormInputs);
+        $findingTheDomCreatedLis.replaceWith(dataEl);
 
-  $entriesList.prepend($callRender);
+        // finding the LI and changing it's value end
+      }
+    });
+    renderEntry(objStorageFormInputs);
```

Comment:
Instead of saying `replaceWith(dataEl)`, you should be calling `renderEntry` and passing in `objStorageFormInputs` and that's what you would pass into `replaceWith`